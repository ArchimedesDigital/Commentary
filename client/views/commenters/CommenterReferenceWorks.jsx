
CommenterReferenceWorks = React.createClass({

	propTypes: {
		commenter: React.PropTypes.object.isRequired,
	},

	render() {
		const commenter = this.props.commenter;
		return (
			<div
				className="commenter-reference-works"
			>
				<div className="commenter-reference-works-title">
					<h2>
						Reference Works
					</h2>
				</div>
				<ReferenceWorksList
					commenterId={commenter._id}
				/>
			</div>
		);
	},

});
