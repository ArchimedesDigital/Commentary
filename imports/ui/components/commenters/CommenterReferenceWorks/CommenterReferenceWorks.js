import React from 'react';

import ReferenceWorksList from '/imports/ui/components/referenceWorks/ReferenceWorksList';

const CommenterReferenceWorks = React.createClass({

	propTypes: {
		commenter: React.PropTypes.object.isRequired,
	},

	render() {
		const { commenter } = this.props;
		if (!commenter) {
			return null;
		}
		
		return (
			<div
				className="commenter-reference-works"
			>
				<div className="commenter-reference-works-title">
					<h2>
						Reference Works
					</h2>
				</div>
				<ReferenceWorksList
					commenterId={commenter._id}
				/>
			</div>
		);
	},

});


export default CommenterReferenceWorks;
