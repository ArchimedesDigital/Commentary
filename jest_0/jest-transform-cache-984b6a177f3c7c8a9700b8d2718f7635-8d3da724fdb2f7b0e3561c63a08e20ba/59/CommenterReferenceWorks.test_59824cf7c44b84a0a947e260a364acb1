/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_reactTestRenderer = require('react-test-renderer'); /* istanbul ignore next */var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);


var /* istanbul ignore next */_CommenterReferenceWorks = require('./CommenterReferenceWorks'); /* istanbul ignore next */var _CommenterReferenceWorks2 = _interopRequireDefault(_CommenterReferenceWorks);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('CommenterReferenceWorks', function () {
	it('renders correctly', function () {

		var tree = /* istanbul ignore next */_reactTestRenderer2.default.
		create(
		/* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_CommenterReferenceWorks2.default, /* istanbul ignore next */{
			commenter: {
				_id: 'testId',
				name: 'Test Commenter',
				slug: 'test-commenter' },

			settings: {
				title: 'Test Settings' },

			isTest: /* istanbul ignore next */true })).


		toJSON();
		expect(tree).toMatchSnapshot();
	});
}); // component: