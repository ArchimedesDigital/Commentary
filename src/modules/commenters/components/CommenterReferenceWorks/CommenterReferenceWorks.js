import React from 'react';
import PropTypes from 'prop-types';

import ReferenceWorksList from '../../../referenceWorks/ReferenceWorksList';

const CommenterReferenceWorks = ({ commenter }) => (
	<div
		className="commenter-reference-works"
	>
		<div className="commenter-reference-works-title">
			<h2>
				Reference Works
			</h2>
		</div>
		<ReferenceWorksList
			commenterId={commenter._id}
		/>
	</div>
);
CommenterReferenceWorks.propTypes = {
	commenter: PropTypes.shape({
		_id: PropTypes.string.isRequired,
	}).isRequired,
};


export default CommenterReferenceWorks;
