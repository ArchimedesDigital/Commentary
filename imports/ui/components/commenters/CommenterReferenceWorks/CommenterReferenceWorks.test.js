import React from 'react';
import renderer from 'react-test-renderer';

// component:
import CommenterReferenceWorks from './CommenterReferenceWorks';

describe('CommenterReferenceWorks', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(
				<CommenterReferenceWorks
					commenter={{
						name: 'Test Commenter',
						slug: 'test-commenter'
					}}
				/>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
