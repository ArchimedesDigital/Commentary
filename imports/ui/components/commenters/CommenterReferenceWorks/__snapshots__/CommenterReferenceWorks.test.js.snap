// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommenterReferenceWorks renders correctly 1`] = `
<div
  className="commenter-reference-works"
>
  <div
    className="commenter-reference-works-title"
  >
    <h2>
      Reference Works
    </h2>
  </div>
</div>
`;
